File: wtf-android/src/main/java/wtf/sdk/WtfApi.java
package wtf.sdk;

//@see SimpleHybridWebViewUi

abstract public class WtfApi {
    private WtfUi __callerUi = null;

    public WtfUi getCallerUi() {
        return __callerUi;
    }

    public void setCallerUi(WtfUi callerUi) {
        __callerUi = callerUi;
    }

    //@NOTES: diff with iOS, but have the same meaning
    //abstract public void handler(JSO jso, WtfCallback cbFunc);
    abstract public WtfHandler getHandler();
}

File: wtf-ios/WtfApi.m
#import "WtfApi.h"

@implementation WtfApi

- (WtfHandler) getHandler{
    return ^(JSO *ddd, WtfCallback responseCallback) {
        dispatch_async(dispatch_get_main_queue(), ^{
            if(nil!=responseCallback) responseCallback(ddd);
        });
    };
}

@end
